actionview/lib/action_view/renderer/partial_renderer/collection_caching.rb

```diff
@@ -51,7 +51,7 @@ def collection_by_cache_keys
       end
 
       def expanded_cache_key(key)
-        key = @view.fragment_cache_key(@view.cache_fragment_name(key))
+        key = @view.fragment_cache_key(@view.cache_fragment_name(key, virtual_path: @template.virtual_path))
```

Comment:
I believe the `virtual_path` is only set to the context once the template is rendered. So when `fragment_name_with_digest ` is called from the partial using `cache`, it has the `@virtual_path` set.